kernel/src/main/scala/cats/kernel/instances/set.scala

```diff
@@ -19,8 +24,16 @@ class SetPartialOrder[A] extends PartialOrder[Set[A]] {
     else if (x == y) 0.0
     else Double.NaN
 
-  override def eqv(x: Set[A], y: Set[A]): Boolean =
-    x == y
+  // Does not require an Eq on elements: Scala sets must use the universal `equals`.
+  override def eqv(x: Set[A], y: Set[A]): Boolean = x == y
+}
+
+class SetHash[A] extends Hash[Set[A]] {
```

Comment:
don't you want to merge this with `SetPartialOrder[A]` since both extend `Eq[A]` otherwise we can have ambiguous implicits?

Reply:
There are a precedence relation in SetInstances and SetInstances1. The precedence of `Hash` is higher than `PartialOrder`, so when trying to resolve an `Eq`, `Hash` would be found.

Reply:
Can you add that comment? I'm not sure why what you said is a problem.